aten/src/ATen/native/cpu/PowKernel.cpp

```diff
@@ -0,0 +1,151 @@
+#include <cmath>
+#include <ATen/Dispatch.h>
+#include <ATen/Parallel.h>
+#include <ATen/cpu/vec256/vec256.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/Pow.h>
+#include <ATen/native/cpu/Loops.h>
+
+namespace at { namespace native {
+
+namespace {
+
+void pow_tensor_tensor_kernel(TensorIterator& iter) {
+  if (isFloatingType(iter.dtype())) {
+    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "pow", [&]() {
+      using Vec = Vec256<scalar_t>;
+      cpu_kernel_vec(iter,
+        [=](scalar_t base, scalar_t exp) -> scalar_t {
+          return std::pow(base, exp);
+        },
+        [&](Vec base, Vec exp) -> Vec {
+          return base.pow(exp);
+        }
+      );
+    });
+  } else {
+    AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "pow", [&]() {
+      cpu_kernel(iter,
```

Comment:
Please add .pow stub from the vec256_base.h in order to make floating and integral dispatch similar.

Reply:
Addressed

Reply:
Trying to implement pow/sqrt/rsqrt as loops in vec256_int.h does not allow powering integral tensor to float exponent. That's why we need this code duplication